test/test_alias.cpp

```diff
@@ -299,6 +299,33 @@ TEST_F(AliasTest, View) {
   testValidate(fec.fusion(), {out_tensor}, {in_tensor}, __LINE__, __FILE__);
 }
 
+TEST_F(AliasTest, View_AliasForCompatibleLayout) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  const std::vector<int64_t> in_shape({2, 3, 4});
+  const std::vector<int64_t> out_shape({2, 12});
+
+  TensorView* in = makeContigConcreteTensor(in_shape);
+  fusion->addInput(in);
+  TensorView* out = reshape(in, in_shape, out_shape);
+  fusion->addOutput(out);
+
+  in->setAllocationDomain(
+      {in->axis(1), in->axis(2), in->axis(0)}, {true, false, false});
+  out->setAllocationDomain({out->axis(1), out->axis(0)}, false);
+
+  FusionExecutorCache fec(std::move(fusion));
+  at::Tensor in_tensor =
+      at::randn({60}).cuda().as_strided({2, 3, 4}, {2, 20, 5});
+  std::vector<at::Tensor> out_tensors = fec.runFusionWithInputs({in_tensor});
+  ASSERT_EQ(out_tensors.size(), 1);
+  at::Tensor out_tensor = out_tensors[0];
+  testValidate(fec.fusion(), {out_tensor}, {in_tensor}, __LINE__, __FILE__);
+
+  EXPECT_TRUE(out_tensor.is_alias_of(in_tensor));
```

Comment:
Nice!

cc'ing @kevinstephano since you were asking about support like this earlier.